Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -134,14 +134,32 @@ private func parseBytes(
         let digit = source[index]
         if digit == asciiDoubleQuote {
             break
-        } else if digit == asciiPlus || digit == asciiForwardSlash {
-            sawSection4Characters = true
-        } else if digit == asciiMinus || digit == asciiUnderscore {
-            sawSection5Characters = true
-        }
-        let k = base64Values[Int(digit)]
-        if k >= 0 {
-            rawChars += 1
+        } else if digit == asciiBackslash {
+            source.formIndex(after: &index)
+            if index == end {
+                throw JSONDecodingError.malformedString
+            }
+            let escaped = source[index]
+            switch escaped {
+            case asciiLowerU:
+                // TODO: Caller should pre-parse with full string decoder and then try again
+                throw JSONDecodingError.malformedString
+            case asciiForwardSlash:
+                rawChars += 1
```

Comment:
Should you set `sawSection4Characters = true` here? Why not just reassign digit and fall through to the else?

Reply:
Oops, yup, seems like it should have been done.

Reply:
Good catch!  I've fixed this and added a test for it.